### DISCUSSION TITLE
执行ddl工单报错

### DISCUSSION THREAD

**Discussion by nishizhuaheihei** [hhyo/Archery#1729] [Q&A]
配置ddl使用gh-ost来做，然后ddl失败，报错信息如下：
<img width="790" alt="image" src="https://user-images.githubusercontent.com/57478779/182508915-12e24b12-ce08-4123-938f-60b0d9f13b69.png">
goInception日志如下：
time="2022/08/02 23:42:21.204" level=info msg="con:62 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:42:21.561" level=info msg="con:62 use binary gh-ost: alter table * engine = innodb" file=session_inception.go func=executeRemoteStatement line=1441
time="2022/08/02 23:42:24.937" level=info msg="con:63 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:42:24.939" level=info msg="con:63 close connection" file=server.go func=func1 line=321
time="2022/08/02 23:42:27.113" level=info msg="con:64 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:42:27.115" level=info msg="con:64 close connection" file=server.go func=func1 line=321
time="2022/08/02 23:42:28.018" level=info msg="con:65 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:42:28.019" level=info msg="con:65 close connection" file=server.go func=func1 line=321
time="2022/08/02 23:42:28.772" level=info msg="con:66 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:42:28.773" level=info msg="con:66 close connection" file=server.go func=func1 line=321
time="2022/08/02 23:42:29.953" level=info msg="con:67 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:42:29.955" level=info msg="con:67 close connection" file=server.go func=func1 line=321
time="2022/08/02 23:42:30.580" level=info msg="con:68 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:42:30.581" level=info msg="con:68 close connection" file=server.go func=func1 line=321
time="2022/08/02 23:42:31.123" level=info msg="con:69 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:42:31.124" level=info msg="con:69 close connection" file=server.go func=func1 line=321
time="2022/08/02 23:42:31.701" level=info msg="con:70 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:42:31.702" level=info msg="con:70 close connection" file=server.go func=func1 line=321
time="2022/08/02 23:43:58.699" level=info msg="con:71 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:43:58.702" level=info msg="con:71 close connection" file=server.go func=func1 line=321
time="2022/08/02 23:44:00.482" level=info msg="con:72 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:44:00.483" level=info msg="con:72 close connection" file=server.go func=func1 line=321
time="2022/08/02 23:44:01.264" level=info msg="con:73 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:44:01.266" level=info msg="con:73 close connection" file=server.go func=func1 line=321
time="2022/08/02 23:44:03.287" level=info msg="con:74 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:44:03.289" level=info msg="con:74 close connection" file=server.go func=func1 line=321
time="2022/08/02 23:44:04.234" level=info msg="con:75 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:44:04.235" level=info msg="con:75 close connection" file=server.go func=func1 line=321
time="2022/08/02 23:44:06.735" level=info msg="con:76 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:44:06.737" level=info msg="con:76 close connection" file=server.go func=func1 line=321
time="2022/08/02 23:44:07.960" level=info msg="con:77 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:44:07.961" level=info msg="con:77 close connection" file=server.go func=func1 line=321
time="2022/08/02 23:44:08.639" level=info msg="con:78 new connection ***:***" file=server.go func=onConn line=319
time="2022/08/02 23:44:08.640" level=info msg="con:78 close connection" file=server.go func=func1 line=321
time="2022/08/03 01:04:02.653" level=error msg="sh [-c gh-ost --alter \"ENGINE = innodb\"  --assume-master-host= --exact-rowcount=false --concurrent-rowcount=true --allow-on-master=true --allow-master-master=false --allow-nullable-unique-key=false --approve-renamed-columns=true --tungsten=false --discard-foreign-keys=false --skip-foreign-key-checks=true --aliyun-rds=false --gcp=false --ok-to-drop-table=false --initially-drop-old-table=true --initially-drop-ghost-table=true --cut-over=atomic --force-named-cut-over=false --assume-rbr=true --cut-over-exponential-backoff=false --exponential-backoff-max-interval=64 --chunk-size=1000 --dml-batch-size=10 --default-retries=60 --cut-over-lock-timeout-seconds=3 --nice-ratio=0.000000 --max-lag-millis=1500 --replication-lag-query= --heartbeat-interval-millis=500 --throttle-flag-file= --throttle-additional-flag-file= --postpone-cut-over-flag-file= --initially-drop-socket-file=false --serve-socket-file=/tmp/gh-ost.***.***.***.sock --panic-flag-file=/tmp/gh-ost.***.***.***.panic --replica-server-id=*** --critical-load-interval-millis=0 --critical-load-hibernate-seconds=0 --critical-load='Threads_running=80,threads_connected=1000'  --max-load='Threads_running=80,threads_connected=1000'  --execute  --user=\"*\" --password='*' --host=* --port=* --database='*' --table='*']" file=osc.go func=execCommand line=902
time="2022/08/03 01:04:02.654" level=error msg="exit status 1" file=osc.go func=execCommand line=903
time="2022/08/03 01:04:02.674" level=info msg="con:62 close connection" file=server.go func=func1 line=321

**Comment by hhyo** [comment]
看起来使用的是外部调用的方式，可以复制相关命令到goinception部署环境执行看具体错误，如果是docker部署需要到容器内执行，问题复现可以反馈给相关项目
